### PR TITLE
fix(condo): DOMA-9137 do not load all billing accounts and billing properties

### PR DESCRIPTION
**Pull request by dkoviazin**
In case Organization has a big number of accounts or properties

### CODE REVIEW

File: apps/condo/domains/billing/schema/resolvers/accountResolver.js [hunk 1]
```diff
@@ -28,7 +28,12 @@ class AccountResolver extends Resolver {
     }
 
     async init () {
-        this.accounts = await find('BillingAccount', { context: { id: this.billingContext.id }, deletedAt: null })
+        //this.accounts = await find('BillingAccount', { context: { id: this.billingContext.id }, deletedAt: null })
```
Comment (sitozzz) [inline apps/condo/domains/billing/schema/resolvers/accountResolver.js:31]: Just delete the line if you don't need it
```suggestion
```

**Review by ekabardinsky — APPROVED** [review]

File: apps/condo/domains/billing/schema/resolvers/accountResolver.js [hunk 2]
```diff
@@ -28,7 +28,12 @@ class AccountResolver extends Resolver {
...
+    }
+
+    async getBillingAccount (where = {}) {
+        const [account] = await find('BillingAccount', { ...where, deletedAt: null, context: { id: this.billingContext.id } })
```
Comment (AleX83Xpert) [inline apps/condo/domains/billing/schema/resolvers/accountResolver.js:35]:
```suggestion
        const [account] = await find('BillingAccount', { ...where, deletedAt: null, context: { id: this.billingContext.id, deletedAt: null } })
```

File: apps/condo/domains/billing/schema/resolvers/accountResolver.js [hunk 3]
```diff
@@ -60,26 +65,26 @@ class AccountResolver extends Resolver {
...
+                existingAccount = await this.getBillingAccount( { number: accountNumber, property: { id: receipt.property } })
             }
             if (!existingAccount) {
-                const sameNumberAccount = this.accounts.find(({ number }) => number === accountNumber)
+                const sameNumberAccount = await this.getBillingAccount( { number: accountNumber })
                 if (sameNumberAccount) {
                     const oldBillingProperty = await getById('BillingProperty', sameNumberAccount.property)
-                    const [organizationProperty] = await find('Property', { addressKey: oldBillingProperty.addressKey, deletedAt: null, organization: { id: get(this.billingContext, 'organization.id') } })
-                    if (!organizationProperty) {
+                    const [oldOrganizationProperty] = await find('Property', { addressKey: oldBillingProperty.addressKey, deletedAt: null, organization: { id: get(this.billingContext, 'organization.id') } })
```
Comment (AleX83Xpert) [inline apps/condo/domains/billing/schema/resolvers/accountResolver.js:77]:
```suggestion
                    const [oldOrganizationProperty] = await find('Property', { addressKey: oldBillingProperty.addressKey, deletedAt: null, organization: { id: get(this.billingContext, 'organization.id'), deletedAt: null } })
```

File: apps/condo/domains/billing/schema/resolvers/propertyResolver.js [hunk 4]
```diff
@@ -27,21 +26,43 @@ class PropertyResolver extends Resolver {
...
         this.parser = new AddressFromStringParser()
         this.tin = get(billingContext, 'organization.tin')
+        this.organizationId = get(billingContext, 'organization.id')
         this.transform.init(get(billingContext, 'settings.addressTransform', {}))
         this.isCottageVillage = !!(get(billingContext, 'settings.isCottageVillage'))
         this.addressService = createInstance()
     }
 
+    async getBillingProperty (where = {}) {
+        const [property] = await find('BillingProperty', { ...where, context: { id: this.billingContext.id }, deletedAt: null })
```
Comment (AleX83Xpert) [inline apps/condo/domains/billing/schema/resolvers/propertyResolver.js:40]:
```suggestion
        const [property] = await find('BillingProperty', { ...where, context: { id: this.billingContext.id, deletedAt: null }, deletedAt: null })
```

File: apps/condo/domains/billing/schema/resolvers/propertyResolver.js [hunk 5]
```diff
@@ -27,21 +26,43 @@ class PropertyResolver extends Resolver {
...
+
+    async getOrganizationProperty (addressKey) {
+        if (!this.organizationId) {
+            return null
+        }
+        const loaded = this.organizationProperties.find(({ addressKey: propertyAddressKey }) => propertyAddressKey === addressKey)
+        if (loaded) {
+            return loaded
+        }
+        const [property] = await find('Property', { addressKey, organization: { id: this.organizationId }, deletedAt: null })
```
Comment (AleX83Xpert) [inline apps/condo/domains/billing/schema/resolvers/propertyResolver.js:52]:
```suggestion
        const [property] = await find('Property', { addressKey, organization: { id: this.organizationId, deletedAt: null }, deletedAt: null })
```

File: apps/condo/domains/billing/schema/resolvers/propertyResolver.js [hunk 6]
```diff
@@ -27,21 +26,43 @@ class PropertyResolver extends Resolver {
...
+    }
+
     async init () {
-        const organizationProperties = await find('Property', { organization: { id: get(this.billingContext, 'organization.id') }, deletedAt: null })
-        this.organizationProperties = Object.fromEntries(organizationProperties.map(({ addressKey, address }) => ([addressKey, address])))
-        this.properties = await find('BillingProperty', { context: { id: this.billingContext.id }, deletedAt: null })
-        this.propertyFinder.init(organizationProperties)
+        if (this.isCottageVillage) {
+            // We still need to load all properties in case we do not use addressService for address normalize
+            const organizationProperties = await find('Property', { organization: { id: this.organizationId }, deletedAt: null })
```
Comment (AleX83Xpert) [inline apps/condo/domains/billing/schema/resolvers/propertyResolver.js:62]:
```suggestion
            const organizationProperties = await find('Property', { organization: { id: this.organizationId, deletedAt: null }, deletedAt: null })
```

**Review by AleX83Xpert — APPROVED** [review]

File: apps/condo/domains/billing/schema/resolvers/propertyResolver.js (hunk 6 above)
Comment (dkoviazin) [reply apps/condo/domains/billing/schema/resolvers/propertyResolver.js:62]: We have organization and context as an input, so we already check that organization and context are not deleted
properties, accounts, receipts - are checked to not to be deleted

File: apps/condo/domains/billing/schema/resolvers/accountResolver.js (hunk 1 above)
Comment (dkoviazin) [reply apps/condo/domains/billing/schema/resolvers/accountResolver.js:31]: will remove it later, needs this to make some tests

**Pull request merged**